events/include/events/EventQueue.h

```diff
@@ -89,30 +89,28 @@ class EventQueue : private mbed::NonCopyable<EventQueue> {
     /** Dispatch events
      *
      *  Executes events until the specified milliseconds have passed.
-     *  If ms is negative, the dispatch function will dispatch events
-     *  indefinitely or until break_dispatch is called on this queue.
      *
-     *  When called with a finite timeout, the dispatch function is guaranteed
-     *  to terminate. When called with a timeout of 0, the dispatch function
-     *  does not wait and is IRQ safe.
+     *  The dispatch function is guaranteed to terminate after the elapsed wait
      *
-     *  @param ms       Time to wait for events in milliseconds, a negative
-     *                  value will dispatch events indefinitely
-     *                  (default to -1)
+     *  @param wait     Time to wait for events in milliseconds, expressed as a 
+     *                  Chrono duration.
      */
-    void dispatch(int ms = -1);
+    void dispatch(duration wait);
```

Comment:
You need to leave the existing API with a deprecation warning. This PR then becomes a Feature update rather than Breaking change.